invenio_records_resources/records/systemfields/files/manager.py

```diff
@@ -360,7 +360,7 @@ def sync(self, src_files, delete_extras=True):
                 if obj_or_key in self:
                     del self[obj_or_key]
             elif operation == "add":
-                self[obj_or_key.key] = obj_or_key
+                self[obj_or_key.key] = obj_or_key, dict(src_files[obj_or_key.key])
```

Comment:
just to understand what is happening: why do you need to cast it to dict? what type does `src_files[obj_or_key.key]` return ?

Reply:
I updated the code a bit more to understand, essentially, I am trying to replicate exactly what's happening here: https://github.com/inveniosoftware/invenio-records-resources/blob/47bf68f6eaca88f004f7f642827552fc77edf021/invenio_records_resources/records/systemfields/files/manager.py#L324-L328

`src_files[obj_or_key.key]` <- gives the latest file metadata